receiver/postgresqlreceiver/client_mock_test.go

```diff
@@ -1,112 +1,137 @@
-	metricStats := []*MetricStat{}
-	metricStats = append(metricStats, &MetricStat{
-		database: c.database,
-		table:    "public.table1",
-		stats:    map[string]string{"live": "7", "dead": "8"},
-	})
-	metricStats = append(metricStats, &MetricStat{
-		database: c.database,
-		table:    "public.table2",
-		stats:    map[string]string{"live": "9", "dead": "10"},
-	})
-	return metricStats, nil
+func (c *fakeClient) getDatabaseSize(databases []string) ([]MetricStat, error) {
+	metrics := []MetricStat{}
+	for idx, db := range databases {
+		metrics = append(metrics, MetricStat{
+			database: db,
+			stats:    map[string]string{"db_size": fmt.Sprintf("%d", idx+4)},
+		})
+	}
+
+	return metrics, nil
 }
 
-func (c *fakeClient) getBlocksReadByTable() ([]*MetricStat, error) {
-	metricStats := []*MetricStat{}
-	metricStats = append(metricStats, &MetricStat{
+func (c *fakeClient) getDatabaseTableMetrics() ([]MetricStat, error) {
+	idx := 0
+	for i, db := range c.databases {
+		if db == c.database {
+			idx = i
+			break
+		}
+	}
+	metrics := []MetricStat{}
+	metrics = append(metrics, MetricStat{
 		database: c.database,
 		table:    "public.table1",
 		stats: map[string]string{
-			"heap_read":  "19",
-			"heap_hit":   "20",
-			"idx_read":   "21",
-			"idx_hit":    "22",
-			"toast_read": "23",
-			"toast_hit":  "24",
-			"tidx_read":  "25",
-			"tidx_hit":   "26",
+			"live":    fmt.Sprintf("%d", idx+7),
+			"dead":    fmt.Sprintf("%d", idx+8),
+			"ins":     fmt.Sprintf("%d", idx+39),
+			"upd":     fmt.Sprintf("%d", idx+40),
+			"del":     fmt.Sprintf("%d", idx+41),
+			"hot_upd": fmt.Sprintf("%d", idx+42),
 		},
 	})
-	metricStats = append(metricStats, &MetricStat{
+
+	metrics = append(metrics, MetricStat{
 		database: c.database,
 		table:    "public.table2",
 		stats: map[string]string{
-			"heap_read":  "27",
-			"heap_hit":   "28",
-			"idx_read":   "29",
-			"idx_hit":    "30",
-			"toast_read": "31",
-			"toast_hit":  "32",
-			"tidx_read":  "33",
-			"tidx_hit":   "34",
+			"live":    fmt.Sprintf("%d", idx+9),
+			"dead":    fmt.Sprintf("%d", idx+10),
+			"ins":     fmt.Sprintf("%d", idx+43),
+			"upd":     fmt.Sprintf("%d", idx+44),
+			"del":     fmt.Sprintf("%d", idx+45),
+			"hot_upd": fmt.Sprintf("%d", idx+46),
 		},
 	})
-	return metricStats, nil
+
+	return metrics, nil
 }
 
-func (c *fakeClient) getOperationsByTable() ([]*MetricStat, error) {
-	metricStats := []*MetricStat{}
-	metricStats = append(metricStats, &MetricStat{
+func (c *fakeClient) getBlocksReadByTable() ([]MetricStat, error) {
+	idx := 0
+	for i, db := range c.databases {
+		if db == c.database {
+			idx = i
+			break
+		}
+	}
+	metrics := []MetricStat{}
+	metrics = append(metrics, MetricStat{
 		database: c.database,
 		table:    "public.table1",
 		stats: map[string]string{
-			"ins":     "39",
-			"upd":     "40",
-			"del":     "41",
-			"hot_upd": "42",
+			"heap_read":  fmt.Sprintf("%d", idx+19),
+			"heap_hit":   fmt.Sprintf("%d", idx+20),
```

Comment:
small thing but wouldn't be easier to just do something like
```suggestion
			"heap_hit":   fmt.Sprintf("%d", idx++),
```

Reply:
I don't think so, I'm trying to have it be pretty clear that the number is the base on the right, plus whatever number database we're on. Wanted data to be distinct between DBs for the mocking so it would be different data but didn't want to have to make the mock client a lot larger